lib/myapp18_web/user_auth.ex

```diff
@@ -114,13 +134,38 @@ defmodule Myapp18Web.UserAuth do
       conn = fetch_cookies(conn, signed: [@remember_me_cookie])
 
       if token = conn.cookies[@remember_me_cookie] do
-        {token, put_token_in_session(conn, token)}
+        {token, conn |> put_token_in_session(token) |> put_session(:user_remember_me, true)}
       else
-        {nil, conn}
+        nil
       end
     end
   end
 
+  # Reissue the session token if it is older than the configured reissue age.
+  defp maybe_reissue_user_session_token(conn, user, token_inserted_at) do
+    token_age = DateTime.diff(DateTime.utc_now(), token_inserted_at, :day)
+
+    if token_age >= @session_reissue_age_in_days do
+      new_token = Accounts.generate_user_session_token(user)
+
+      conn
+      |> put_token_in_session(new_token)
+      |> maybe_refresh_remember_me_cookie(new_token)
```

Comment:
Note to self: this is very close to the login function, we should probably break the login function in two, one which logs in and the other that logs in and redirect. But we want to make other changes to that function, so we will do this later. :) No action needed from you.